### ISSUE TITLE
AVA fails when running test files which are actually symlinks 

### ISSUE THREAD

**Issue by novemberborn** [avajs/ava#1143]
Per #1137, if a test pattern matches a symlink, AVA fails to run that test file.

AVA precompiles the test file in the main process. That works fine, even with symlinks. It then communicates a hash for the precompilation result to the worker process. When the test file is required the worker loads it from the precompilation cache, using the hash received from the main process.

The problem is that the hash mapping is for the symlink path, but the worker sees the _resolved_ path instead.

This is where the worker looks up the hash: https://github.com/avajs/ava/blob/0f0400158ec4192fc32ccc603877d0b088302662/lib/process-adapter.js#L85

This is where the `precompiled` hash mapping is prepared: https://github.com/avajs/ava/blob/0f0400158ec4192fc32ccc603877d0b088302662/api.js#L70

Probably the best fix is to use [`fs.realpathSync`](https://nodejs.org/api/fs.html#fs_fs_realpathsync_path_options) when preparing the hash mapping in `api.js`.

**Comment by rnkdev** [comment]
Hi guys, this is my very first time I actually did something to an open source program. 

So I followed the hint in the issue tracker and modified api.js so that the hash now uses resolved path as the key. (I hope this is as expected and I'm just not breaking everything) 

I ran `npm test` before committing and it passed all of the tests. 

I don't want to create pull request because I'm not sure if this is the right way to do it or not. 

https://github.com/avajs/ava/commit/7a1a0cdebbe870650ff02956a41a78ea2cc41a34
